com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -57,49 +57,94 @@ internal struct AgentAction
     }
 
     /// <summary>
-    /// Agent MonoBehaviour class that is attached to a Unity GameObject, making it
-    /// an Agent. An agent produces observations and takes actions in the
-    /// environment. Observations are determined by the cameras attached
-    /// to the agent in addition to the vector observations implemented by the
-    /// user in <see cref="Agent.CollectObservations(VectorSensor)"/>.
-    /// On the other hand, actions are determined by decisions produced by a Policy.
-    /// Currently, this class is expected to be extended to implement the desired agent behavior.
+    /// An agent is an actor that can observe its environment, decide on the
+    /// best course of action using those observations, and execute those actions
+    /// within the environment. 
     /// </summary>
     /// <remarks>
-    /// Simply speaking, an agent roams through an environment and at each step
-    /// of the environment extracts its current observation, sends them to its
-    /// policy and in return receives an action. In practice,
-    /// however, an agent need not send its observation at every step since very
-    /// little may have changed between successive steps.
+    /// Create an agent by extending the Agent class. Add your Agent subclass to
```

Comment:
Sorry, this is a bit out of date - it's not strictly necessary to subclass `Agent` anymore (although in general you probably want to, and almost all of out examples do).

Reply:
So what do you want to say here?

Reply:
I would remove the whole paragraph, unless you think it's necessary to mention the GameObject and Scene.

Reply:
The general idea is to capture the gist of how to create an Agent. 

Even though, technically, you could rewrite a slightly different class that does the same thing as Agent, wouldn't you be essentially rewriting much or most of the same functionality? Unless you want to instead document how to implement your own base class in place of Agent, I'd say this was fine (it doesn't claim to be the only way to create an agent).

While I think it unnecessary, we could rewrite as:

"Use the Agent class as the base class for implementing your own agents."

Which describes what the Agent class is for without seeming as prescriptive, but really, it is a more wordy way of saying essentially the same thing.

Reply:
> "Use the Agent class as the base class for implementing your own agents."

That sounds good!